src/NodeApi/Runtime/NativeLibrary.cs

```diff
@@ -44,12 +44,50 @@ public static nint GetMainProgramHandle()
     public static nint Load(string libraryName)
     {
 #if NETFRAMEWORK || NETSTANDARD
-        return LoadLibrary(libraryName);
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            return LoadLibrary(libraryName);
+        }
+        else
+        {
+            var h = dlopen(libraryName, RTLD_LAZY);
+            if (h == 0)
+                throw new DllNotFoundException();
+
+            return h;
+        }
 #else
         return SysNativeLibrary.Load(libraryName);
 #endif
     }
 
+    /// <summary>
+    /// Provides a simple API for loading a native library and returns a value that indicates whether the operation succeeded.
+    /// </summary>
+    /// <param name="libraryName">The name of the native library to be loaded.</param>
+    /// <param name="handle">When the method returns, the OS handle of the loaded native library.</param>
+    /// <returns><c>true</c> if the native library was loaded successfully; otherwise, <c>false</c>.</returns>
+    public static bool TryLoad(string libraryName, out nint handle)
```

Comment:
Can we implement the `Load` by using the `TryLoad`? It is to reduce the code duplication.

Reply:
NativeLibrary in dotnet/runtime does this through a LoadFromFilePath method that takes a 'throwOnError' argument. Which I added.